src/components/GlobalAudioPlayer/MediaController.tsx

```diff
@@ -0,0 +1,211 @@
+'use client'
+
+import React, { useState, useEffect, useRef } from 'react'
+import { useAccount } from 'wagmi'
+
+import { useMedia } from '@/context/MediaContext'
+import MediaSheet from '../Sheet/Mediasheet'
+import { Icons } from '@/components/Icons'
+import { formatTime } from '@/lib/utils'
+import { PlayMode } from '@/lib/enum'
+
+export const MediaController: React.FC = () => {
+  const intoClass =
+    'w-full fixed bottom-0 dark:bg-black bg-white text-white p-3 z-50 border-t-[1px] border-zinc-500'
+  const iconClass = 'w-4 h-4 text-zinc-400 dark:hover:text-white hover:text-black'
+  const [refresh, setRefresh] = useState<number>(0)
+  const audioRef = useRef<HTMLAudioElement>(null)
+  const audio = audioRef.current
+  const { isConnected } = useAccount()
+  const {
+    currentMedia,
+    medias,
+    isPlaying,
+    setIsPlaying,
+    setCurrentMedia,
+    playStatus,
+    setPlayStatus,
+    handleSongEnded,
+  } = useMedia()
+
+  useEffect(() => {
+    const audio = audioRef.current
+    if (audio) {
+      audio.src = medias[currentMedia]?.url
+    }
+  }, [currentMedia, medias, audio])
+
+  useEffect(() => {
+    if (audio) {
+      const handleLoadedMetadata = () => {
+        if (isPlaying) {
+          audio.play()
+        }
+      }
+
+      const handleTimeUpdate = () => {
+        setRefresh(prev => prev + 1)
+      }
+
+      const handleEnded = () => {
+        setRefresh(prev => prev + 1)
+        handleSongEnded()
+      }
+
+      audio.addEventListener('loadedmetadata', handleLoadedMetadata)
+      audio.addEventListener('timeupdate', handleTimeUpdate)
+      audio.addEventListener('ended', handleEnded)
+
+      return () => {
+        audio.removeEventListener('loadedmetadata', handleLoadedMetadata)
+        audio.removeEventListener('timeupdate', handleTimeUpdate)
+        audio.removeEventListener('ended', handleEnded)
+      }
+    }
+  }, [audio, medias, currentMedia, handleSongEnded])
```

Comment:
can you move this `useEffect` out of the react component into a new hook?

maybe name it `useMediaController`?